### PR TITLE
feat: CFX Maestro - initial implementation

### PR DESCRIPTION
**Pull request by Preston-O-Benchling**

### CODE REVIEW

**Comment by Preston-O-Benchling** [comment]
Note: this connector will not include calculated data as the replicate number field was not exported with the example file and the calculates cannot be made without it. The structure that was excluded because of this include:
Calculated Data Aggregate Document
Calculated Data Source Aggregate Document
Data Source  Aggregate Document
Data Source Document
Data Processing Document

A future update will be required were the replicate number is included in the example file to include these pieces.

**Comment by nathan-stender** [comment]
@james-leinas to review ASM output

**Comment by nathan-stender** [comment]
@slopez-b @alejandro-salgado I approve this code, but I made some contributions so need another reviewer, if one of you could take a look!

File: src/allotropy/parsers/cfxmaestro/constants.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from allotropy.allotrope.models.adm.pcr.benchling._2023._09.qpcr import ContainerType
+
+DISPLAY_NAME = "CFX Maestro"
```
Comment (james-leinas) [inline src/allotropy/parsers/cfxmaestro/constants.py:3]: @Preston-O-Benchling, can we change the display name of the adapter to Bio-Rad CFX Maestro so we are capturing the company name that produces the software.

**Review by slopez-b — DISMISSED** [review]

File: tests/parsers/cfxmaestro/testdata/example01.json [hunk 2]
```diff
@@ -0,0 +1,8467 @@
...
+                                        "data processing document": {
+                                            "cycle threshold value setting": {
+                                                "value": -0.0,
+                                                "unit": "(unitless)"
+                                            }
+                                        },
+                                        "cycle threshold result": {
+                                            "value": null,
+                                            "unit": "(unitless)"
+                                        }
```
Comment (james-leinas) [inline tests/parsers/cfxmaestro/testdata/example01.json:755]: @Preston-O-Benchling, what are your thoughts on skipping of wells that contain no `Sample Name` and no `Cq` value? These effectively are empty wells that contain no results data. You have better context if this is enough information to designate whether the well is empty however.

**Review by james-leinas — CHANGES_REQUESTED** [review]
Requesting change for display name and question regarding recording of empty wells

File: tests/parsers/cfxmaestro/testdata/example01.json (hunk 2 above)
Comment (Preston-O-Benchling) [reply tests/parsers/cfxmaestro/testdata/example01.json:755]: Great call! Lets only exclude values if they do not have both a sample name and no Cq value. This way if the user accidentally forgets to label a well or mixes up the well plate, the value wont be excluded.

Comment (nathan-stender) [reply tests/parsers/cfxmaestro/testdata/example01.json]: Done!

**Review by slopez-b — APPROVED** [review]

**Review by james-leinas — APPROVED** [review]
ASM output LGTM 🎉

**Review by stephenworlow — APPROVED** [review]

**Comment by nathan-stender** [comment]
Re-uploaded here: https://github.com/Benchling-Open-Source/allotropy/pull/744 with signed commits

**Pull request closed without merging**